fastwarc/fastwarc/cli.py

```diff
@@ -167,13 +169,87 @@ def check(infile, decompress_alg, verify_payloads, quiet, output):
                     click.echo(rec)
                 sys.exit(1)
 
+@main.command()
+@click.argument('infile', type=click.Path(dir_okay=False, exists=True))
+@click.argument('offset', type=click.INT)
+@click.option('--payload', is_flag=True,
+              help='output only record payload (transfer and/or content encoding are preserved')
+@click.option('--headers', is_flag=True, help='output only record (and HTTP) headers')
+def extract(infile, offset, payload, headers):
+    """Extract WARC record by offset.
+    """
+    compl_record = not (payload or headers)
+    with open(infile, 'rb') as stream:
+        stream.seek(offset)
+        record = next(iter(ArchiveIterator(stream)))
```

Comment:
Ok. But I'd had to catch all kinds of exceptions.

Reply:
Not really. Calling `next()` on any kind of iterator would raise `StopIteration`, which needs to be caught. That applies to iterators created with `iter()` as well:

```python
>>> next(iter([]))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
StopIteration
```

Reply:
Hmhh. But I see for example if pass an offset exceeding the WARC file size:
 ```
  ...
  File "fastwarc/warc.pyx", line 966, in fastwarc.warc.ArchiveIterator.__next__
  File "fastwarc/warc.pyx", line 950, in __iter__
  File "fastwarc/warc.pyx", line 969, in fastwarc.warc.ArchiveIterator._read_next_record
  File "fastwarc/stream_io.pyx", line 750, in fastwarc.stream_io.BufferedReader.detect_stream_type
fastwarc.stream_io.StreamError: Not a valid WARC stream
```

Reply:
Ah, right, that's because the stream negotiation cannot determine the compression type. With an explicit compressor, you would get the same exception, but a different message ("Not a valid GZip stream"). Didn't think of that, but that's an error you may actually want to catch and print. 

`StopIteration` would occur if you started at a valid offset and then went one item beyond the end or if the stream had a valid GZip header, but no further content. So in this case it's indeed a somewhat unlikely scenario, but that would be different if you allowed reading of more than one record (which should be rather trivial to implement).